src/JSONSchemasInterface.js

```diff
@@ -2,6 +2,43 @@ import { schemas } from "@exabyte-io/esse.js/schemas";
 import mergeAllOf from "json-schema-merge-allof";
 
 const schemasCache = new Map();
+
+/**
+ * We assume that each schema in the application has its own unique schemaId
+ * Unfortunately, mergeAllOf keeps schemaId after merging, and this results in multiple different schemas with the same schemaId
+ * Hence this function
+ */
+function removeSchemaIdsAfterAllOf(schema, clean = false) {
```

Comment:
Should we add a test for this?

Reply:
The test `can find registered schemas` checks this logic. The test name is not accurate, so I changed it to `can find registered schemas; the schema is merged and clean`